x-pack/plugin/security/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRestIT.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.remotecluster;
+
+import org.apache.http.HttpHost;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.ResponseException;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.common.UUIDs;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.test.rest.ESRestTestCase;
+import org.elasticsearch.test.rest.ObjectPath;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Locale;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+
+/**
+ * This test suite will be run twice: Once against the fulfilling cluster, then again against the querying cluster. The typical usage is to
+ * conditionalize on whether the test is running against the fulfilling or the querying cluster.
+ */
+public class RemoteClusterSecurityRestIT extends ESRestTestCase {
```

Comment:
There are a lot more tests we need (essentially what's covered in `multi-cluster-search-security` + edge-case testing around DLF/FLS and more than one fulfilling cluster) but that requires more test setup and could take quite a long time, so opting to postpone that work to a separate PR. 

The current suite covers basic search and authc + authz failure scenarios.

Reply:
One more test I **do** want to add in this iteration, just didn't have time to yet is that setting the cluster authorization setting to `null` and the proxy address to normal transport port reverts things back to the legacy model.